src/wallet/test/walletdb_tests.cpp

```diff
@@ -29,5 +29,104 @@ BOOST_AUTO_TEST_CASE(walletdb_readkeyvalue)
     BOOST_CHECK_THROW(ssValue >> dummy, std::ios_base::failure);
 }
 
+BOOST_AUTO_TEST_CASE(walletdb_readonly_loadwallet)
```

Comment:
In 6189a20491f71e3a6f499d2a510b38bb914e00d9 "test: Add tests for readonly database access"

As in my previous comment regarding the tests in `db_tests.cpp`, the added tests in this file are not testing anything since `MockableDatabase` is not enforcing the read-only. These tests are definitely more useful conceptually than the `db_tests.cpp` tests since it is testing the call stack from the wallet level. However, I believe these would still pass if anything in that call stack would write things to the database.